src/os/bluestore/BlueStore.cc

```diff
@@ -4774,6 +4797,21 @@ void BlueStore::handle_conf_change(const ConfigProxy& conf,
       changed.count("osd_memory_expected_fragmentation")) {
     _update_osd_memory_options();
   }
+  if (changed.count("bluestore_codel") ||
+      changed.count("bluestore_codel_slow_interval") ||
+      changed.count("bluestore_codel_fast_interval") ||
+      changed.count("bluestore_codel_initial_target_latency") ||
+      changed.count("bluestore_codel_min_target_latency") ||
+      changed.count("bluestore_codel_max_target_latency") ||
+      changed.count("bluestore_codel_throughput_latency_tradeoff") ||
+      changed.count("bluestore_codel_initial_budget_bytes") ||
+      changed.count("bluestore_codel_min_budget_bytes") ||
+      changed.count("bluestore_codel_budget_increment_bytes") ||
+      changed.count("bluestore_codel_regression_history_size")) {
+    if (codel) {
```

Comment:
Shouldn't we create codel instance if bluestroe_codel is set to true and no instance exists? This will allow codel enablement on-the-fly if it wasn't enabled on OSD startup.

Reply:
Is it hard to implement CoDel shutdown on config  change as well?